transport/tchannel/channel_outbound.go

```diff
@@ -253,3 +257,13 @@ func getResponseErrorAndDeleteHeaderKeys(headers transport.Headers) error {
 	errorMessage, _ := headers.Get(ErrorMessageHeaderKey)
 	return intyarpcerrors.NewWithNamef(errorCode, errorName, errorMessage)
 }
+
+func checkServiceMatchAndDeleteHeaderKey(reqSvcName string, resHeaders transport.Headers) (bool, string) {
+	defer func() {
```

Comment:
FYI: we can just do `defer resHeaders.Del(ServiceHeaderKey)`

Reply:
Defer statement is removed.